Classes/EventStore/EventFromStream.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Neos\Cqrs\EventStore;
+
+/*
+ * This file is part of the Neos.EventStore package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+final class EventFromStream
```

Comment:
I like `StoredEvent`.
It's important to know in what "state" the event is. So we have:

### publishing

1. The original `EventInterface` instance is published via `Aggregate::recordThat()`
2. It is wrapped in a `EventWithMetadata` (aka `EventTransport`) object in `Repository::save()`
3. It is converted to a `WritableEvent` in `EventPublisher::publish()` and sent to the EventStore
4. The `EventWithMetadata` (aka `EventTransport`) is sent to the `EventBus::handle()`
5. The `EventListenerInterface::when*()` methods are called with the original `EventInterface` instance + optional `EventMetadata`

### reconstitution

1. `Repository::findByIdentifier()` loads the `EventStream`  which is basically an `Iterator` of `StoredEvent` (aka `EventFromStream`) instances that are lazily converted to `EventWithMetadata` in `EventStream::current()`
2. The `EventWithMetadata` is passed to `Aggregate::apply()`

### projection replay

1. An `EventStream` is loaded from the EventStore
2. The original `EventInterface` & `EventMetadata` is extracted from the `EventWithMetadata` and passed to the `when*()` method of the projector